SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -163,7 +165,10 @@
   <!-- **************************************************************************** -->
   <!-- Calculate the set of files to be analyzed -->
   <!-- **************************************************************************** -->
-  <Target Name="CalculateSonarQubeFilesToAnalyze" AfterTargets="CoreCompile" BeforeTargets="RunCodeAnalysis"
+  <!-- This target is a utility target: it won't be executed unless a target that
+       depends on it is executed.
+  -->
+  <Target Name="CalculateSonarQubeFilesToAnalyze"
```

Comment:
These two targets (SonarQubeCategoriseProject and CalculateSonarQubeFilesToAnalyze) are helper targets. It doesn't matter exactly when these targets are run, except that they must be run before any of our targets that use the properties they set (the "consuming" targets).
Rather than trying to specify when these helper targets should be run using "Before/AfterTargets", it's simpler and clearer to make the consuming targets explicitly refer to them as dependencies. MSBuild will then make sure they are invoked when they are needed.